### PR TITLE
fix(registry): update -config to -deployment in /deployments

### PR DESCRIPTION
**Pull request by brohamgoham**
### Description
Update Configuration File Naming Convention from *-config.yaml to *-deployment.yaml
This PR is in reference to [hyperlane-monorepo issue #4199 ](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4199
)

- [x] Changed the suffix from *-config.yaml to *-deployment.yaml when warp deployments are written to the registry's /deployments directory

- [x] Updated any associated references and documentation to ensure consistency with the new naming convention.
- [x] Update file names in /deployments

### Testing

- [x] Run registry unit tests

### CODE REVIEW

File: test/unit/registry.test.ts
Comment (brohamgoham) [inline test/unit/registry.test.ts:1]: Registry Unit test fails with 1 error. 
Occurs when fetching github Registry
 `Fetches warp route configs for github registry`

**Review by jmrossy — COMMENTED** [review]
Thanks @brohamgoham for the PR but this change would break compatibility with the existing Registry tooling deployed to NPM. Any users of the existing packages (including the Hyperlane core team), would have the warp route config fetching broken by the name change. Can I ask why you'd like to rename them? They're called `configs` because they contain the configs needed to create a `WarpCore`, the main class used to interact with warp routes

**Comment by brohamgoham** [comment]
I was looking to take on the task in issue #4199. I apologize must have misunderstood the task, will delete now

**Comment by jmrossy** [comment]
@brohamgoham Ah I see. I think maybe the specs in that ticket need some refinement first. Thanks for taking the initiative. We can find you something else ready for implementation if you'd like something.

**Comment by brohamgoham** [comment]
@jmrossy yes please, I would love to contribute!

**Pull request closed without merging**